schema/snapcraft.yaml

```diff
@@ -199,6 +199,25 @@ properties:
             description: What kind of wrapper to generate for the given command
             enum:
               - none
+          sockets:
+            type: object
+            additionalProperties: false
+            patternProperties:
+              "^[a-zA-Z0-9](?:-?[a-zA-Z0-9])*$":
+                type: object
+                required:
+                  - listen-stream
+                description: Sockets for automatic service activation
+                additionalProperties: false
+                properties:
+                  listen-stream:
+                    anyOf:
+                      - type: integer
+                        minimum: 1
+                        maximum: 65535
+                      - type: string
```

Comment:
As it is, let's say I use a value that doesn't match this schema, say, `listen-stream: true`. That will result in the following dreadfully unhelpful error:

```
Issues while validating snapcraft.yaml: The 'apps/foo/sockets/bar/listen-stream' property does not match the required schema: True is not valid under any of the given schemas
```

This is of course not your fault, but it's something we're trying to improve. For example, if you change it to e.g.

```
properties:
  listen-stream:
    anyOf:
      - type: integer
        usage: "port number, an integer between 1 and 65535"
        minimum: 1
        maximum: 65535
      - type: string
        usage: "socket path, a string"
```

There error message is immediately more helpful:

```
Issues while validating snapcraft.yaml: The 'apps/foo/sockets/bar/listen-stream' property does not match the required schema: True is not valid under any of the given schemas (must be one of 'port number, an integer between 1 and 65535' or 'socket path, a string')
```

Reply:
Nice, I didn't know about this. changed as suggested